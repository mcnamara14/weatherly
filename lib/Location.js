import React, { Component } from 'react';
import Trie from 'TDD';
import data from './cities'
import key from './key.js';

const trie = new Trie();
trie.populate(data);

export default class Location extends Component {
  constructor() {
    super();

    this.state = {
      userInputLocation: '',
      suggestions: [],
    };
  }

  setLocation(location) {
    this.setState({suggestions: trie.suggest(location)});
    this.setState({userInputLocation: location});
  }

  render() {
    return(
      <section className="locationInput">
      <span>
        <input 
          list="cities"
          value={this.state.userInputLocation}
          onChange={event => {
            if(event.target.value.length === 5 && /\d/.test(event.target.value)) {
              fetch(`http://api.wunderground.com/api/${key}/geolookup/q/${event.target.value}.json`)
              .then(data => data.json())
              .then(data => {
                this.setLocation(`${data.location.city}, ${data.location.state}`);
              });
            }
            this.setLocation(event.target.value);
          }}
          placeholder='+ Enter Location'
          onKeyPress={target => {
            let city = this.state.userInputLocation.slice(0, location.length-4);
            let state = this.state.userInputLocation.slice(location.length-2, location.length);
            if(target.charCode === 13) {
              trie.select(this.state.userInputLocation);
              this.props.updateLocation(this.state.userInputLocation);
              this.props.updateWeather(city, state);
              this.props.toggleWelcome();
            }
          }} />
        <datalist id="cities">
          {this.state.suggestions.map(city => <option value={city} /> )}
        </datalist>
      </span>
      <button className="submitBtn" onClick={() => {
              this.props.updateLocation(this.state.userInputLocation);
              this.props.toggleWelcome();
        }}>Submit</button>
      </section>
    )
  }
}

