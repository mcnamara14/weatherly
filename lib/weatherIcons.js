const weatherIcons = {
  images: {
    'Clear': 'wu wu-black wu-256 wu-clear',
    'Mostly Cloudy': 'u wu-white wu-256 mostly-cloudy',
    'Overcast': 'wu wu-white wu-256 wu-cloudy', 
    'Snow': 'wu wu-white wu-256 wu-snow',
    'Fog': 'wu wu-white wu-256 wu-fog',
    'Rain': 'wu wu-white wu-256 wu-rain',
    'Light Drizzle': 'wu wu-white wu-256 wu-rain',
    'Mostly Cloudy': 'wu wu-white wu-256 wu-mostlycloudy',
    'Scattered Clouds': 'wu wu-white wu-256 wu-mostlycloudy',
    'Mostly Sunny': 'wu wu-white wu-256 wu-mostlysunny',
    'Chance of Flurries': 'wu wu-white wu-256 wu-chanceflurries',
    'Chance of Rain': 'wu wu-white wu-256 wu-chancerain',
    'Chance Freezing Rain': 'wu wu-white wu-256 wu-chancesleet',
    'Chance of Snow': 'wu wu-white wu-256 wu-chancesnow	',
    'Chance of Thunderstorms': 'wu wu-white wu-256 wu-chancetstorms',
    'Cloudy': 'wu wu-white wu-256 wu-cloudy',
    'Flurries': 'wu wu-white wu-256 wu-flurries',
    'Haze': 'wu wu-white wu-256 wu-hazy',
    'Partly Cloudy': 'wu wu-white wu-256 wu-partlycloudy',
    'Partly Sunny': 'wu wu-white wu-256 wu-partlysunny',
    'Freezing Rain': 'wu wu-white wu-256 wu-sleet',
    'Ice Pellets': 'wu wu-white wu-256 wu-sleet',
    'Snow': 'wu wu-white wu-256 wu-snow',
    'Snow Showers': 'wu wu-white wu-256 wu-snow',
    'Sunny': 'wu wu-white wu-256 sunny',
    'Partly Sunny': 'wu wu-white wu-256 wu-partlysunny',
    'Thunderstorm': 'wu wu-white wu-256 wu-tstorms',
    'Chance of a Thunderstorm': 'wu wu-white wu-256 wu-tstorms',
    'Uknown': 'wu wu-white wu-256 wu-unknown',
  }
}

module.exports = weatherIcons;
