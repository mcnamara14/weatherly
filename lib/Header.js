import React, { Component } from 'react';
import Location from './Location';

export default class Header extends Component {
  constructor() {
    super();
  
  }

    render() {
        return (
            <header>
                <div>
                <img className="logo" src="./assets/weatherly-logo.png"/>
                <Location
                  updateLocation={this.props.updateLocation}
                  updateWeather={this.props.updateWeather}
                  welcomeStatus={this.props.welcome}/>
                 </div>
            </header>
        )
    }
}