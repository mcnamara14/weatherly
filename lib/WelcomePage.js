import React, { Component } from 'react';
import Location from './Location';

export default class WelcomePage extends Component {
  constructor() {
    super();
  }

    render() {
        return (
          <main>
            <section className="welcomeBox">
                <div>
                  <h1>Weathrly</h1>
                  <p>Enter a location to view the forecast</p>
                  <Location updateLocation={this.props.updateLocation} 
                  updateWeather={this.props.updateWeather}
                  toggleWelcome={this.props.toggleWelcome}/>
                 </div>
            </section>
          </main>
        )
    }
}