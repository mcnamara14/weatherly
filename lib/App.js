import React, { Component } from 'react';
import Header from './Header';
import cleanedData from './mock-data';
import WelcomePage from './WelcomePage';
import CurrentWeather from './CurrentWeather';
import SevenHour from './SevenHour';
import TenDay from './TenDay';
import cleanData from './cleanData';
import key from './key.js';
import "isomorphic-fetch";
import weatherUndergroundIcons from 'weather-underground-icons';
import { images } from './weatherIcons';

export default class App extends Component {
  constructor() {
    super();

    this.state = {
      city: '',
      stateUSA: '',
      data: cleanedData || {},
      welcome: true
		};

		this.images = images;
		this.toggleWelcome = this.toggleWelcome.bind(this);
		this.updateLocation = this.updateLocation.bind(this);
	}

	componentDidMount() {
		this.getLocalStorage();
	}

	getLocalStorage() {
		if (window.localStorage.location) {
			this.toggleWelcome();
			var retrievedLocation = 
				JSON.parse(localStorage.getItem('location'));

			this.updateWeather(retrievedLocation.city, 
												retrievedLocation.stateUSA);
		}
	}

	updateLocation(location) {
		let city = location.slice(0, location.length - 4);
		let state = location.slice(location.length - 2, location.length);
		let inputtedLocation = {
			city,
			stateUSA: state
		};
		let stringifiedLocation = JSON.stringify(inputtedLocation);
		
		localStorage.setItem('location', stringifiedLocation);
		this.setState({city, stateUSA: state});
		this.updateWeather(city, state);
	}

	toggleWelcome() {
		this.setState({
			welcome: false
		});
	}

	updateWeather(city, state) {
		fetch(`http://api.wunderground.com/api/${key}/forecast/
					hourly/forecast10day/conditions/q/${state}/${city}.json`)
		.then(data => data.json())
		.then(data => {
			let cleanedData = cleanData(data);

			this.setState({
				data: cleanedData
			});
		}).catch(() => {
			const errorContainer = document.querySelector('.errorContainer');
			const errorBox = document.querySelector('.errorBox');

			errorContainer.classList.toggle('hide');
			errorBox.classList.toggle('hide');
		});
	}

	render() {
		if (this.state.welcome === true) {
			return (
				<WelcomePage updateLocation={this.updateLocation} 
											toggleWelcome={this.toggleWelcome}/>
			);    
		} else {
			return (
				<main>
				<Header updateLocation={this.updateLocation} />
				<section className="container">
					<CurrentWeather
						images={this.images}
						current={this.state.data.current}
						forecast={this.state.data.forecast}
					/>
					<SevenHour 
						images={this.images}
						currentCondition=
							{this.state.data.current.currentCondition}
						hourly={this.state.data.hourly}
					/>
					<TenDay 
						images={this.images}
						forecast={this.state.data.forecast.tenDayArray}
					/>
				</section>
				</main>
			);}
	}
}