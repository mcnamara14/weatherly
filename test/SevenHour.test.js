import React from 'react';
import { shallow } from 'enzyme';
import SevenHour from '../lib/SevenHour.js';
import cleanedData from '../lib/mock-data.js';
import { images } from '../lib/weatherIcons';

describe('Seven hour', () => {

  let renderedSevenHour = shallow(<SevenHour 
    images={images}
    currentCondition= {cleanedData.current.curentCondition}
    hourly={cleanedData.hourly}
  />);

    it('should render text inside div with class todaysWeather', () => {
      const expectation = "";

      expect(renderedSevenHour.find('.hourTemp')
      .text()).toEqual(expectation);
    });
});