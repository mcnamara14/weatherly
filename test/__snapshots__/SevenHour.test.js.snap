// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seven hour Seven hout component renders the seven hour correctly should render correctly 1`] = `
<div>
  <section
    className="extendedForecastHeader"
  >
    <hr />
    <h3>
      7-Hour
      <br />
       
      <span>
        Forecast
      </span>
    </h3>
    <hr />
  </section>
  <section
    className="sevenHourForecast"
  >
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        11
        <span>
          AM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          64
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        12
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          64
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        1
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          65
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        2
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          63
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        3
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          62
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        4
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          63
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        5
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          62
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
  </section>
</div>
`;

exports[`Seven hour Todo component renders the todo correctly renders correctly 1`] = `
<div>
  <section
    className="extendedForecastHeader"
  >
    <hr />
    <h3>
      7-Hour
      <br />
       
      <span>
        Forecast
      </span>
    </h3>
    <hr />
  </section>
  <section
    className="sevenHourForecast"
  >
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        11
        <span>
          AM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          64
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        12
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          64
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        1
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          65
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        2
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          63
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        3
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          62
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        4
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          63
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
    <section
      className="hourForecast"
    >
      <article
        className="extendedTimeDay"
      >
        5
        <span>
          PM
        </span>
      </article>
      <div>
        <article
          className="hourTemp"
        >
          62
          <sup>
            o
          </sup>
        </article>
        <i
          className="wu wu-white wu-256 wu-mostlycloudy"
        />
      </div>
    </section>
  </section>
</div>
`;
